File: 1137. N-th Tribonacci Number/dp.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n == 0) return 0;
        else if (n == 1 || n == 2) return 1;
        int arr[4] = {0, 1, 1, 0};
        for (int i = 3; i <= n; i++) {
            arr[3] = arr[1] + arr[2] + arr[0];
            arr[0] = arr[1];
            arr[1] = arr[2];
            arr[2] = arr[3];
        }
        return arr[3];
    }
};

File: 1137. N-th Tribonacci Number/dp.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        elif n == 1 or n == 2:
            return 1
        res = [0] * (4)
        res[0] = 0
        res[1] = 1
        res[2] = 1
        for i in range(3, n + 1):
            res[3] = res[2] + res[1] + res[0]
            res[0] = res[1]
            res[1] = res[2]
            res[2] = res[3]
        return res[3]